File: app/src/main/java/online/fatbook/fatbookapp/ui/listeners/OnRecipeClickListener.java
package online.fatbook.fatbookapp.ui.listeners;

import online.fatbook.fatbookapp.core.Recipe;

public interface OnRecipeClickListener {
    void onRecipeClick(int position);
    void onBookmarksClick(Recipe recipe, boolean bookmark, int position);
    void onForkClicked(Recipe recipe, boolean fork, int position);
}

File: app/src/main/java/online/fatbook/fatbookapp/ui/listeners/OnRecipeClickListener.kt
package online.fatbook.fatbookapp.ui.listeners

import online.fatbook.fatbookapp.core.recipe.RecipeSimpleObject
import online.fatbook.fatbookapp.ui.adapters.RecipeAdapter

interface OnRecipeClickListener {
    fun onRecipeClick(id: Long)
    fun onBookmarksClick(recipe: RecipeSimpleObject?, bookmark: Boolean, position: Int)
    fun onForkClicked(recipe: RecipeSimpleObject?, fork: Boolean, position: Int)
    fun onForkClicked(
        recipe: RecipeSimpleObject?,
        fork: Boolean,
        position: Int,
        viewHolder: RecipeAdapter.ViewHolder
    )
    fun onUsernameClick(username: String)

}